### DISCUSSION TITLE
EntitySchema custom entity classes do not work if class is defined in seperate file

### DISCUSSION THREAD

**Discussion by Fatih-Ertikin** [mikro-orm/mikro-orm#5662] [General]
### Describe the bug

[According to the docs](https://mikro-orm.io/docs/entity-schema#using-custom-entity-classes), you can use custom classes to define your entities with entitySchema. The example works but when you define the class in a seperate file, mikro-orm can't seem to find the class.

Error: `Source class for entity [entityName] not found. Verify you have 'compilerOptions.declaration' enabled in your 'tsconfig.json'. If you are using webpack see https://bit.ly/35pPDNn`.

declaration is enabled in `tsconfig.json`.

### Reproduction

[Link to reproducible bug repo ](https://github.com/Fatih-Ertikin/mikro-orm-bug/tree/master)(I am using nestjs).

`mySeperateClass.ts`:
```typescript
export class User {
  public readonly id: string;
  public readonly name: string;

  constructor(id: string, name: string) {
    this.id = id;
    this.name = name;
  }
}
```

`schema.entity.ts`:
```typescript
import { EntitySchema } from "@mikro-orm/core";
import { User } from "./mySeperateClassFile";

export const schema = new EntitySchema<User>({
  class: User,
  properties: {
    id: { type: "varchar", primary: true },
    name: { type: "varchar" },
  },
});
```

error when starting project (`npm run start:dev`):
![image](https://github.com/mikro-orm/mikro-orm/assets/47319376/aa5bce56-9397-4f3c-9ca9-016fd1efed87)

### What driver are you using?

@mikro-orm/mysql

### MikroORM version

6.2.7

### Node.js version

20.10.0

### Operating system

WSL2

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
You are not supposed to be using ts-morph with this approach, there is no point in doing that since you specify all the metadata explicitly anyway.

**Reply by Fatih-Ertikin** [reply]
Hi, Thanks for the quick reply :).

Yeah TSMorph was setup by the nestjs example repo. I tought that it leaving it would have no impact for the bug repo.
Or would you like me to remove tsmorph?

**Reply by Fatih-Ertikin** [reply]
Okay I think i didn't test this well enough... removing tsmorph seems to fix the issue.

**Comment by Fatih-Ertikin** [comment]
Issue fixed by removing tsmorph from the config
